nixos/modules/services/web-apps/lemmy.nix

```diff
@@ -216,10 +224,23 @@ in
 
         requires = lib.optionals cfg.database.createLocally [ "postgresql.service" ];
 
+        path = mkIf (cfg.secretFile != null) [ pkgs.jq ];
+
+        # merge the two configs and prevent others from reading the result
+        preStart = mkIf (cfg.secretFile != null) ''
+          set -u
+          umask 177
+          jq --slurp '.[0] * .[1]' "${configFile}" "$CREDENTIALS_DIRECTORY/secretFile" > "${mergedConfig}"
+        '';
+
         serviceConfig = {
           DynamicUser = true;
           RuntimeDirectory = "lemmy";
           ExecStart = "${cfg.server.package}/bin/lemmy_server";
+          LoadCredential = mkIf (cfg.secretFile != null) "secretFile:${cfg.secretFile}";
+          PrivateTmp = true; # important to prevent the mergend config from being replaced
+          MemoryDenyWriteExecute = true;
+          NoNewPrivileges = true;
```

Comment:
Since you have added some hardening options, it would be good if you try to add more of them.  FYI, `systemd-analyze security lemmy.service` is useful for this.  This is not needed for this PR to be merged though.

Reply:
An example: https://github.com/NixOS/nixpkgs/blob/1c9db9710cb23d60570ad4d7ab829c2d34403de3/nixos/modules/services/hardware/kanata.nix#L105

Reply:
I added the hardening options from the [example](https://join-lemmy.org/docs/administration/from_scratch.html), before adding additional options I would like to have the tests working again (https://github.com/NixOS/nixpkgs/issues/236890).